### PR TITLE
Search updates, adding daily-image feature 

### PR DESCRIPTION
**Pull request by RobertKelly**
Decided to add another feature so we could test/discuss this one in action...
- `/daily-image` gets today's day name and searches for a meme/image with that in mind (Wednesday stays camel focused)
- `/daily-image some text` will just go ahead and do a search with that text... why not

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,5 +78,18 @@ With all of these steps complete, your bot should be running in the Slack
 workspace you chose to develop for. You should now be ready to test your bot,
 and progress with development.
 
+#### Creating a Slash Command
+1. Under *Slash Commands* select *Create New Command*
+2. Add the slash command you wish to use - this can be anything you wish to add as a message you want your bot to respond to.
+3. Add a short description and save your changes.
```
Comment (Pactionly) [inline README.md]: Consider adding some details about what request url to use.

File: features/daily-image.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+module.exports = function(controller) {
+ 
+  controller.on('slash_command', async (bot, message) => {
+    console.log(message);
+    if (message.command == '/daily-image') {
+      let imageSearchString = getImageQueryString(message.text.trim());
+      console.log('Image Search String = '+imageSearchString);
+      //This is an interesting workaround to handle timing of the nested promises
+      await bot.changeContext(message.reference);
```
Comment (Pactionly) [inline features/daily-image.js]: Is changeContext() async? Is that related to the workaround being described?

File: features/daily-image.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      await getImage(bot, message, imageSearchString);
+    }
+  });
+ };
+
+async function getImage(bot, message, imageSearchString) {
+  let searchUrl = 'https://api.bing.microsoft.com/v7.0/images/search?q='+imageSearchString+'&size=medium&mkt=en-us&count=50'
+  await axios.get(searchUrl, {
```
Comment (Pactionly) [inline features/daily-image.js]: It seems to me like this should require an await _and_ a .then() block. Is there a reason to use both of them here?

**Review by Pactionly — APPROVED** [review]

File: features/daily-image.js (hunk 2 above)
Comment (RobertKelly) [reply features/daily-image.js]: I believe the bot is now getting a response within its request timeout to set its original context so any chained "slow" responses won't be revoked.

File: features/daily-image.js (hunk 3 above)
Comment (RobertKelly) [reply features/daily-image.js]: Ah, I think it is redundant. Good catch.

**Review by Pactionly — APPROVED** [review]
LGTM
